docs/guides/caching.md

```diff
@@ -0,0 +1,117 @@
+# Caching
+
+When using the recommended approach for initializing experiences with this SDK, the logic will take fetch the experience entry, other entries, assets, and patterns under the hood and return the final experience object.
+
+To inject a manual caching layer in between, the SDK allows you to fetch the content on your own.
+
+## Typical Fetching & Rendering
+
+The simple setup would usually look something like this calling one of the sophisticated fetcher functions `useFetchBySlug`. While the final experience object exposes all fetched content, it doesn't provide a way to inject already fetched content but will fetch everything from the API at once.
+
+```ts
+const { experience, error, isLoading } = useFetchBySlug({
+  slug,
+  localeCode: locale,
+  client,
+  experienceTypeId,
+});
+```
+
+## Caching & Injecting Data
+
+To cache the loaded entities and inject them manually, a custom fetcher function needs to be written that uses `fetchExperienceEntry`, `fetchReferencedEntities`, and `createExperience` which are exposed from the SDK as well.
+
+### Fetching Data
+
+```ts
+function customFetchAllEntities({ client, experienceTypeId, slug, localeCode }) {
+  const experienceEntry = await fetchExperienceEntry({
+    client,
+    experienceTypeId,
+    identifier: { slug },
+    locale: localeCode,
+  });
+  const references = await fetchReferencedEntities({
+    client,
+    experienceEntry,
+    locale: localeCode,
+  });
```

Comment:
These references also include entries in `usedComponents`? I think so but just confirming.

Reply:
Used Components are fetched together with the experience as we're using `include=3` in there. This one solely cares about dataSource.